src/iuse.cpp

```diff
@@ -6032,15 +6032,36 @@ int iuse::cut_log_into_planks(player *p, item *it)
     }
     p->i_add_or_drop(plank, planks);
     p->i_add_or_drop(scrap, scraps);
-    return it->type->charges_to_use();
+
+    if (it != nullptr) // Should only be nullptr if access through construction menu
+        return it->type->charges_to_use();
+    else
+        return 0;
 }
 
 int iuse::lumber(player *p, item *it, bool, point)
 {
-    int pos = g->inv_for_filter( _("Cut up what?"), []( const item & itm ) {
-        return itm.type->id == "log";
-    } );
-    item *cut = &( p->i_at( pos ) );
+    item* cut = nullptr;
+
+    // Check if player is standing on any lumber
+    for (auto &i : g->m.i_at(p->posx(), p->posy())) {
+        if (i.type->id == "log")
+        {
+            cut = &i;
```

Comment:
Storing the pointer in `cut` is pointless, just remove the item:
```C++
g->m.i_rem( ..., &i )
```

`cut` is not needed at all until it is used to store the result of `g->inv_for_filter` (like it was done before the changes).

Below the test for `cut == nullptr` will *always* be true: the only place before that where `cut` has been set to something, is right before the function is left with the `return`. Even if the function was not left, `cut` will point to an non-existing thing, the item it pointed to was removed via `g->m.remove_item`.

About `cut_log_into_planks`:
I recommend removing the `it` parameter completely and return nothing at all. The caller does either not care about the charges that was returned (for constructions) or the caller already has the item and can get the charges from it. This avoids the usage of the null pointer.